src/Microsoft.ServiceFabric.Actors/Runtime/MigrationActorStateProvider.cs

```diff
@@ -0,0 +1,394 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Actors.Runtime
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Fabric;
+    using System.Fabric.Health;
+    using System.Runtime.Serialization;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Xml;
+    using Microsoft.ServiceFabric.Actors;
+    using Microsoft.ServiceFabric.Actors.Query;
+    using Microsoft.ServiceFabric.Data;
+    using Microsoft.ServiceFabric.Data.Collections;
+
+    /// <summary>
+    /// Provides an implementation of <see cref="MigrationActorStateProvider"/> which
+    /// uses <see cref="ReliableCollectionsActorStateProvider"/> to store and persist the actor state.
+    /// </summary>
+    public class MigrationActorStateProvider :
```

Comment:
Name it something KVStoRCMigrationActorStateProvider to be explicit

Reply:
Updated